pkg/bootstrap/docker/up.go

```diff
@@ -410,18 +411,31 @@ func (c *ClientStartConfig) EnsureVolumeShare(io.Writer) error {
 
 // CheckAvailablePorts ensures that ports used by OpenShift are available on the Docker host
 func (c *ClientStartConfig) CheckAvailablePorts(out io.Writer) error {
-	err := c.OpenShiftHelper().TestPorts()
-	if err != nil {
-		return errors.NewError("A port needed by OpenShift is not available").WithCause(err)
+	err := c.OpenShiftHelper().TestPorts(openshift.DefaultPorts)
+	if err == nil {
+		c.DNSPort = openshift.DefaultDNSPort
+		return nil
 	}
-	return nil
+	if !openshift.IsPortsNotAvailableErr(err) {
+		return err
+	}
+	conflicts := openshift.UnavailablePorts(err)
+	if len(conflicts) == 1 && conflicts[0] == openshift.DefaultDNSPort {
+		err = c.OpenShiftHelper().TestPorts(openshift.PortsWithAlternateDNS)
+		if err == nil {
+			c.DNSPort = openshift.AlternateDNSPort
+			fmt.Fprintf(out, "WARNING: Using alternate port (%d) for DNS.\nLookups may not work on node.\n", openshift.AlternateDNSPort)
```

Comment:
See what we print in master startup

On May 18, 2016, at 11:52 PM, Cesar Wong <notifications@github.com> wrote:

In pkg/bootstrap/docker/up.go
<https://github.com/openshift/origin/pull/8935#discussion_r63819552>:

>  	}
> -	return nil
> +	if !openshift.IsPortsNotAvailableErr(err) {
> +		return err
> +	}
> +	conflicts := openshift.UnavailablePorts(err)
> +	if len(conflicts) == 1 && conflicts[0] == openshift.DefaultDNSPort {
> +		err = c.OpenShiftHelper().TestPorts(openshift.PortsWithAlternateDNS)
> +		if err == nil {
> +			c.DNSPort = openshift.AlternateDNSPort
> +			fmt.Fprintf(out, "WARNING: Using alternate port (%d) for DNS.\nLookups may not work on node.\n", openshift.AlternateDNSPort)

Not sure the wording for this warning is right

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/8935/files/47e064d721623eca8b13957a6e75a50da37a4f97#r63819552>